Automatic live-ish resize engages late; one frame after the live resize starts

hortont424 opened the pull request:
#### e2f0d3ff7df51131fadb3cb0be5d6a00bdd6c90f
<pre>
Automatic live-ish resize engages late; one frame after the live resize starts
<a href="https://bugs.webkit.org/show_bug.cgi?id=247028">https://bugs.webkit.org/show_bug.cgi?id=247028</a>
rdar://101562669

Reviewed by Aditya Keerthi.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView setFrame:]):
(-[WKWebView setBounds:]):
(-[WKWebView layoutSubviews]):
(-[WKWebView _frameOrBoundsWillChange]):
(-[WKWebView _frameOrBoundsMayHaveChanged]):
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
(-[WKWebView _frameOrBoundsChanged]): Deleted.
`_frameOrBoundsChanged` currently engages snapshotted live-ish resize mechanism
*after* updating the frame, which means that it starts one frame behind the
live resize, and so the first frame delta is lost, leading to the right edge
of the content tearing away from the right edge of the WKWebView by the amount
of the delta.

Introduce `_frameOrBoundsWillChange` (called *before* we call super), and rename
`_frameOrBoundsChanged` to `_frameOrBoundsMayHaveChanged` since it is called
in multiple cases where we do not actually know that there was a change.

Canonical link: <a href="https://commits.webkit.org/255992@main">https://commits.webkit.org/255992@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9be7a65ba5ec1dfae089d901f033b2b1ba57c258

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94252 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3439 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/25621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/103928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3469 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/31653 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/86603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99947 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99910 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/80658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/86603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/84411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/25621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/86603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38036 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/25621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35921 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/25621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4145 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39800 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/80658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41744 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/25621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Not a big fan of the name `_frameOrBoundsMayHaveChanged` in all cases, since it's frequently called when we know that the frame/bounds have in fact changed, but don't have a better idea at the moment.

hortont424 committed:
Automatic live-ish resize engages late; one frame after the live resize starts
https://bugs.webkit.org/show_bug.cgi?id=247028
rdar://101562669

Reviewed by Aditya Keerthi.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView setFrame:]):
(-[WKWebView setBounds:]):
(-[WKWebView layoutSubviews]):
(-[WKWebView _frameOrBoundsWillChange]):
(-[WKWebView _frameOrBoundsMayHaveChanged]):
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
(-[WKWebView _frameOrBoundsChanged]): Deleted.
`_frameOrBoundsChanged` currently engages snapshotted live-ish resize mechanism
*after* updating the frame, which means that it starts one frame behind the
live resize, and so the first frame delta is lost, leading to the right edge
of the content tearing away from the right edge of the WKWebView by the amount
of the delta.

Introduce `_frameOrBoundsWillChange` (called *before* we call super), and rename
`_frameOrBoundsChanged` to `_frameOrBoundsMayHaveChanged` since it is called
in multiple cases where we do not actually know that there was a change.

Canonical link: https://commits.webkit.org/255992@main
